app/src/project/yuma_simulation/_internal/yumas.py

```diff
@@ -133,12 +138,99 @@ def binary_search_compute_consensus_weights(W: torch.Tensor, S: torch.Tensor, co
         C[i] = c_high
     return C
 
+def rust_parity_compute_consensus_weights(W: torch.Tensor, S: torch.Tensor, config) -> torch.Tensor:
```

Comment:
I think you found a "bug" in subtensor - where "bug" means - someone implemented somthing correclty - but should actually implement different thing :)